src/stores/game.js

```diff
@@ -140,31 +165,42 @@ export const useGameStore = defineStore('game', {
       if (Object.hasOwnProperty.call(newGame, 'id')) this.id = newGame.id;
       if (Object.hasOwnProperty.call(newGame, 'turn')) this.turn = newGame.turn;
       if (Object.hasOwnProperty.call(newGame, 'chat')) this.chat = cloneDeep(newGame.chat);
-      if (Object.hasOwnProperty.call(newGame, 'deck')) this.deck = cloneDeep(newGame.deck);
-      if (Object.hasOwnProperty.call(newGame, 'scrap')) this.scrap = cloneDeep(newGame.scrap);
+      if (Object.hasOwnProperty.call(newGame, 'deck'))
+        this.deck = newGame.deck?.map((card) => createGameCard(card));
+      if (Object.hasOwnProperty.call(newGame, 'scrap'))
+        this.scrap = newGame.scrap?.map((card) => createGameCard(card));
       if (Object.hasOwnProperty.call(newGame, 'log')) this.log = cloneDeep(newGame.log);
       if (Object.hasOwnProperty.call(newGame, 'name')) this.name = newGame.name;
       if (Object.hasOwnProperty.call(newGame, 'p0Ready')) this.p0Ready = newGame.p0Ready;
       if (Object.hasOwnProperty.call(newGame, 'p1Ready')) this.p1Ready = newGame.p1Ready;
       if (Object.hasOwnProperty.call(newGame, 'passes')) this.passes = newGame.passes;
-      if (Object.hasOwnProperty.call(newGame, 'players')) this.players = cloneDeep(newGame.players);
+      if (Object.hasOwnProperty.call(newGame, 'players'))
+        this.players = newGame.players.map((player) => ({
+          ...player,
+          hand: player.hand?.map((card) => createGameCard(card)) ?? player.hand,
+          points: player.points?.map((card) => createGameCard(card) ?? player.points),
+          faceCards: player.faceCards?.map((card) => createGameCard(card) ?? player.faceCards),
```

Comment:
There's something funky about these coalesces. As written, they should only trigger if `player.hand` etc is `null` or more likely `undefined` i.e. the player doesn't have a `.hand` at all (not just an empty list). And if that's the case, then we're coalescing to the same undefined value i.e. `player.hand` which we just said was undefined. So I think we can just leave them out out. 

Also the one for hand is currently coalescing outside the .map(), like I said above, but `.points()` and `.faceCards` are coalescing _inside_ the `.map()` which is weirder b/c it'll set an individual card instance to the entire list.

I _think_ we just want
```suggestion
          hand: player.hand?.map((card) => createGameCard(card)),
          points: player.points?.map((card) => createGameCard(card)),
          faceCards: player.faceCards?.map((card) => createGameCard(card)),
```

but we should sync up b/c the type safety is weird